Source/driver/Castro.cpp

```diff
@@ -890,6 +901,32 @@ Castro::initData ()
        }
        enforce_consistent_e(S_new);
 
+       // thus far, we assume that all initialization has worked on cell-centers
+       // (to second-order, these are cell-averages, so we're done in that case).
+       // For fourth-order, we need to convert to cell-averages now.
+       if (fourth_order) {
+         Sborder.define(grids, dmap, NUM_STATE, NUM_GROW);
+         AmrLevel::FillPatch(*this, Sborder, NUM_GROW, cur_time, State_Type, 0, NUM_STATE);
+
+         // note: this cannot be tiled
+         for (MFIter mfi(S_new); mfi.isValid(); ++mfi)
+           {
+             const Box& box     = mfi.validbox();
+             const int* lo      = box.loVect();
+             const int* hi      = box.hiVect();
+
+             const int idx = mfi.tileIndex();
+
+             ca_make_fourth_in_place(BL_TO_FORTRAN_BOX(box),
+                                     BL_TO_FORTRAN_FAB(Sborder[mfi]),
+                                     &idx);
+           }
+
+         // now copy back the averages
+         MultiFab::Copy(S_new, Sborder, 0, 0, NUM_STATE, 0);
```

Comment:
This should use S_new.nGrow() ghost zones.

Reply:
Although I guess strictly unnecessary due to the FillPatch that comes next.

Reply:
Maybe we can skip that next FillPatch on line 898 if we did this 4th order conversion already, since we effectively already did the FillPatch we need.

Reply:
the issue is that the `ca_make_fourth_in_place` only deals with the interior.  It cannot do all of the FAB, because ghost cells are needed for the Laplacian